src/measure_theory/integral/exp_decay.lean

```diff
@@ -0,0 +1,143 @@
+/-
+Copyright (c) 2022 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+
+import measure_theory.integral.interval_integral
+
+import analysis.special_functions.exponential
+import analysis.special_functions.integrals
+import measure_theory.integral.integral_eq_improper
+
+
+/-!
+# Integrals with exponential decay at ∞
+
+As easy special cases of general theorems in the library, we prove the following test
+for integrability:
+
+* `exp_decay.integrable_bigoh_exp`: If `f` is continuous on `[a,∞)`, for some `a ∈ ℝ`, and there
+  exists `b > 0` such that `f(x) = O(exp(-b x))` as `x → ∞`, then `f` is integrable on `(a, ∞)`.
+
+-/
+
+noncomputable theory
+open real interval_integral measure_theory set filter
+
+namespace exp_decay
+
+/-- `exp(-b x)` is integrable on `(a, X]` for any finite `a`, `X`. -/
+lemma exp_neg_finite_integrable (b : ℝ) (a : ℝ) (X : ℝ):
+  integrable_on (λ x : ℝ, exp(-b * x) ) (Ioc a X) :=
+  (continuous_const.mul continuous_id).exp.integrable_on_Ioc
+
+/-- The antiderivative of `exp(-b x)` is what it should be. -/
+lemma exp_neg_hasderiv (b : ℝ) (x : ℝ) (h : 0 < b) :
+  has_deriv_at (λ (y : ℝ), -exp (-b * y) / b) (exp (-b * x)) x :=
+begin
+  have : has_deriv_at (λ (x : ℝ), x) 1 x := has_deriv_at_id x,
+  have : has_deriv_at (λ (x : ℝ), b*x) b x := by simpa only [mul_one] using
+    has_deriv_at.const_mul b this,
+  have : has_deriv_at (λ (x : ℝ), -b*x) (-b) x := by simpa only [neg_mul] using this.neg,
+  have : has_deriv_at (λ (x : ℝ), exp (-b*x)) (exp (-b*x)*(-b)) x := has_deriv_at.exp this,
+  have : has_deriv_at (λ (x : ℝ), -exp (-b*x)) (-(exp (-b*x)*(-b))) x := has_deriv_at.neg this,
+  have := (has_deriv_at.div_const this b),
+  have s : (-(exp (-b * x) * -b) / b) = exp(-b*x),
+  { simp only [neg_mul, mul_neg, neg_neg],
+    rw mul_div_cancel _ (h.ne') },
+  rw s at this, exact this
+end
+
+/-- Integral of `exp(-b x)` over `(a, X)` is bounded as `X → ∞`. -/
+lemma exp_neg_integral_bound {b : ℝ} (a X : ℝ) (h2 : 0 < b):
+  (∫ x in a .. X, exp (-b * x)) ≤ exp(-b*a)/b :=
+begin
+  rw (integral_deriv_eq_sub' (λ x:ℝ, -exp(-b*x)/b ) ),
+  -- goal 1/4: F(X) - F(a) is bounded
+  { simp only [tsub_le_iff_right],
+    rw (neg_div b (exp (-b * a))),
+    rw (neg_div b (exp (-b * X))),
+    rw add_neg_self,
+    rw [neg_le, neg_zero],
+    apply le_of_lt,
+    apply div_pos,
+    apply exp_pos,
+    exact h2 },
+  -- goal 2/4: the derivative of F is exp(-b x)
+  { ext1, exact has_deriv_at.deriv ( exp_neg_hasderiv b x h2 ), },
+  -- goal 3/4: F is differentiable (why isn't this implicit?)
+  { intros x hx, exact has_deriv_at.differentiable_at (exp_neg_hasderiv b x h2 ),},
```

Comment:
Note that `rw` calls can be chained using `[lemma1, lemma2, ...]`. You can still inspect the intermediate goal states, by placing you cursor after the `,`s in the list.
```suggestion
lemma exp_neg_integral_bound {b : ℝ} (a X : ℝ) (h2 : 0 < b) :
  (∫ x in a .. X, exp (-b * x)) ≤ exp (-b*a) / b :=
begin
  rw (integral_deriv_eq_sub' (λ x, -exp(-b*x)/b)),
  -- goal 1/4: F(X) - F(a) is bounded
  { simp only [tsub_le_iff_right],
    rw [(neg_div b (exp (-b * a))), (neg_div b (exp (-b * X))), add_neg_self, neg_le, neg_zero],
    exact (div_pos (exp_pos _) h2).le, },
  -- goal 2/4: the derivative of F is exp(-b x)
  { ext1, exact (has_deriv_at_exp_neg b x h2).deriv, },
  -- goal 3/4: F is differentiable (why isn't this implicit?)
  { intros x hx, exact (has_deriv_at_exp_neg b x h2).differentiable_at, },
```
Again, I take some mild advantage of dot-notation.